docs/tutorials/schema-change.mdx

```diff
@@ -0,0 +1,165 @@
+
+## 1. Prerequisites
+
+- A Xata account ([sign up here](https://console.xata.io))
+- The [Xata CLI](/cli) installed:
+  ```bash
+  curl -fsSL https://xata.io/install.sh | bash
+  ```
+- Access to your Xata PostgreSQL branch (connection string)
+- An initialized Xata project and branch
+
+## 2. Create a Xata project and branch
+
+In the Console, create a new project and then click the **Create main branch** button to create the PostgreSQL instance.
+
+![Create new branch](assets/images/create-new-branch.png)
+
+Since this is a production setup, we recommend using at least 1 replica. It's a good idea to start with a similar instance size as you are running in your current PostgreSQL provider. You will be able to adjust it later.
+
+## 3. Configure the Xata CLI
+
+Authenticate the CLI by running:
+
+```sh
+xata auth login
+```
+
+Initialize the project by running:
+
+```sh
+xata init
+```
+
+You would typically run `xata init` command in your application source directory, or in any directory/repository where you'd like to keep the configuration. It will create a `.xata` folder with several configuration files.
+
+Follow the instructions from the prompt to select the project and branch that you created at the previous step. Choose the same database name as you use in production.
+
+## 4. Prepare a migration directory
+
+Create and initialize a directory for your migrations (e.g., `.xata/migrations`) using the following command:
+
+```sh
+xata roll init --postgres-url `xata branch url`
+```
+
+This command sets up migration tracking on your current Xata branch and prepares the directory for your migration files.
+
+## 5. Create initial tables
+
+First, let's create some sample tables using the SQL from our [getting started guide](/getting-started). Connect to your database using psql:
+
+```sh
+psql `xata branch url`
+```
+
+Then run the following SQL:
+
+```sql
+CREATE TABLE products(id SERIAL PRIMARY KEY,name TEXT,price NUMERIC(7,2));
+CREATE TABLE orders(id SERIAL PRIMARY KEY,created_at TIMESTAMPTZ DEFAULT now());
+CREATE TABLE order_items(order_id INT REFERENCES orders(id),product_id INT REFERENCES products(id),qty INT,PRIMARY KEY(order_id,product_id));
+INSERT INTO products(name,price) SELECT LEFT(md5(i::text),8),(random()*90+10)::numeric(7,2) FROM generate_series(1,10)i;
+WITH o AS (INSERT INTO orders DEFAULT VALUES RETURNING id) INSERT INTO order_items(order_id,product_id,qty) SELECT o.id,pid,(1+floor(random()*3))::int FROM o,(SELECT id pid FROM products ORDER BY random() LIMIT 5)p;
+```
+
+## 6. Create a schema change migration
+
+Now, let's create a migration to add a description column and remove the name column from the products table. Create a new file in your migrations directory (e.g., `.xata/migrations/001_alter_products.sql`) with the following content:
+
+```sql
+ALTER TABLE products ADD COLUMN description TEXT;
+ALTER TABLE products DROP COLUMN name;
+```
```

Comment:
This I'm not sure how it should work. Normally migrations in `.xata/migrations` are YAML files, not sql. But maybe the intention is to run `xata convert`?

Reply:
Something like this perhaps:

```
xata roll convert .xata/migrations/001_alter_products.sql > .xata/migrations/001_alter_products.json
```

I also raised a small issuse: https://xata-hq.slack.com/archives/C05TR0HPQ9Y/p1748521358166269

Reply:
I originally had it as JSON / YAML and reverted. I'll just add commands for creating the actual migration files. I must have been tired when drafting this, I definitely was thinking that pgroll could diff and create the files a la Xata Lite.